c7n/resources/s3.py

```diff
@@ -1316,6 +1421,7 @@ def validate(self):
 
     def process(self, resources):
         enabled = self.data.get('enabled', True)
+        update = self.data.get('update', False)
```

Comment:
rather than introducing another variable, we should just check if the configuration of logging matches the current configuration/state and update automatically if there's a delta when enabled is set.

Reply:
Yeah. I wasn’t sure if this change in behavior was something we wanted, so I added a new variable. If folks have policies that use set-logging and assume it will only update the value if it’s disabled, this would change that.  I’ll remove the update variable.

Reply:
it feels like setting to match the policy is doing the right thing, and is the least surprising thing to do.